### ISSUE TITLE
Fix prediction for CNN

### ISSUE THREAD

**Issue by Cyrof** [Cyrof/Artificial-Intelligence-Autonomous-Systems#8]
# CNN prediction bug

CNN prediction accuracy works only within the model class but not in main.  Would like to fix this if possible. 

-----
**Model class accuracy calculation**
``` python
    def test_model(self, dataloader):
        self.eval()
        correct = 0
        total = 0
        predicted_values = []
        with torch.no_grad():
            for inputs, labels in dataloader:
                outputs = self(inputs)
                _, predicted = torch.max(outputs.data, 1)
                # _, predicted = torch.max(outputs, 1)
                predicted_values.append(predicted)
                total += labels.size(0)
                correct += (predicted == labels).sum().item()
        print(f"Accuracy of the network on the test images: {100 * correct/total}%")
        return predicted_values
```

**Model class predict function**
``` python 
    def predict(self, dataloader):
        self.eval()
        predictions = []
        with torch.no_grad():
            for inputs, _ in dataloader:
                # inputs = inputs.view(inputs.size(0), -1)
                outputs = self(inputs)
                _, predicted = torch.max(outputs.data, 1)
                predictions.extend(predicted.tolist())
        return predictions
```
### Important
When test_model function is called, the accuracy returns 90% +. However when predict function is called to return predicted_values to calculate in main.py the accuracy returns 10% or less. The function used to calculate the accuracy in main.py is shown below. <br>
**Calculate accuracy function**
``` python 
def calculate_acc(model, dataloader):
  model.eval()
  predicted_values = model.predict(dataloader)
  # print(predicted_values)
  correct = 0
  total = 0
  idx = 0
  with torch.no_grad():
    for inputs, labels in dataloader:
      batch_size = labels.size(0)
      preds = predicted_values[idx:idx + batch_size]
      idx += batch_size
      total += batch_size
      correct += (torch.tensor(preds) == labels.cpu()).sum().item()
  acc = 100 * correct / total 
  print(f"Accuracy of the network on the test images: {acc:.2f}%")
```
